### PR TITLE
NO-ISSUE support publish ports and set default net to bridge for mac

### PR DESCRIPTION
**Pull request by michaellevy101**

### CODE REVIEW

**Comment by eranco74** [comment]
> /approve

This isn't openshift repo... :smile: 
You need to actually approve

File: tests/test_runner.py [hunk 1]
```diff
@@ -22,6 +24,18 @@
 ENV_FILE_PATH = '/home/envfile.env'
 
 
+def get_default_net():
```
Comment (eranco74) [inline tests/test_runner.py]: Consider moving this oneliner function to runner and use it in both places

File: skipper/cli.py [hunk 2]
```diff
@@ -18,12 +19,32 @@
 DOCKER_TAG_FOR_CACHE = "cache"
 
 
+def _validate_publish(ctx, param, value):
+    # pylint: disable=unused-argument
+    if value:
+        matcher = compile_expression(r'^(\d+):(\d+)$')
```
Comment (eranco74) [inline skipper/cli.py]: There is also an option to publish a range of ports, e.g.
`docker run -p 4000-6000:4000-6000`

File: skipper/cli.py [hunk 3]
```diff
@@ -18,12 +19,32 @@
...
+
+        for port_forwarding in value:
+            _validate_port(matcher, port_forwarding)
```
Comment (eranco74) [inline skipper/cli.py]:
```suggestion
        for port_mapping in value:
            _validate_port(matcher, port_mapping)
```

File: skipper/cli.py [hunk 4]
```diff
@@ -18,12 +19,32 @@
...
+    return value
+
+
+def _validate_port(matcher, port_forwarding):
+    match = matcher.match(port_forwarding)
+    if not match:
+        raise click.BadParameter("publish need to be in format port:port")
+    host_port, container_port = match.group(1, 2)
+    if not 1 <= int(host_port) <= 65535 or not 1 <= int(container_port) <= 65535:
+        raise click.BadParameter("invalid port number - out of range")
```
Comment (eranco74) [inline skipper/cli.py]: add the bad port mapping to the error message

**Review by eranco74 — COMMENTED** [review]
Looks good.
Please update the readme

File: README.md [hunk 5]
```diff
@@ -169,6 +169,17 @@ Using the above configuration file, we now can run a simplified version of the m
 skipper make tests
 ```
 
+### Published ports
+For `shell`, `run` & `make` commands:
```
Comment (eranco74) [inline README.md:173]: Consider adding this option only works when build-container-net != host

**Review by eranco74 — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (michaellevy101) [reply README.md:173]: docker run command ignored this option and log a WARNING so have nothing to do....

**Pull request merged**